src/modules/weather/weather.js

```diff
@@ -156,11 +169,135 @@ export class Weather {
+        });
+
+        this._createSearchCountryInput();
+        this._createSearchCountrySaveButton();
+
+        this._settingsBody.append(this._setingsFormElement);
+    }
+
+    _createSearchCountryInput() {
+        const countrySearchContainer = createElement('div', { class: 'form-floating col ps-0' });
+        this._countrySearchElement = createElement(
+            'input',
+            {
+                class: 'form-control',
+                list: 'datalistOptions',
+                id: 'countrySearch',
+                placeholder: 'Type to search...',
+                value: localStorage.getItem(WEATHER_COUNTRY_KEY),
+            },
+            { keyup: () => this._getAutocompleteCityList() }
+        );
+        countrySearchContainer.append(this._countrySearchElement);
+        countrySearchContainer.append(
+            createElement(
+                'label',
+                {
+                    for: 'countrySearch',
+                },
+                null,
+                'Country ID'
+            )
+        );
+
+        this._createSettingsDatalistElement(countrySearchContainer);
+        this._setingsFormElement.append(countrySearchContainer);
+    }
+
+    _createSettingsDatalistElement(parent) {
+        this._datalistElement = createElement('datalist', { id: 'datalistOptions' });
+        parent.append(this._datalistElement);
+    }
+
+    async _getAutocompleteCityList() {
+        this._datalistElement.innerHTML = '';
+        if (this._countrySearchElement.value !== '') {
+            const cityListData = await this._api.getCityList(this._countrySearchElement.value);
+            cityListData.forEach((dataElement) => {
+                const cityElement = createElement(
+                    'option',
+                    {
+                        value: dataElement.Key,
+                    },
+                    null,
+                    `${dataElement.LocalizedName}, ${dataElement.AdministrativeArea.LocalizedName}, ${dataElement.Country.LocalizedName}`
+                );
+                this._datalistElement.append(cityElement);
+            });
+        }
+    }
+
+    _createSearchCountrySaveButton() {
+        this._setingsFormElement.append(
+            createElement(
+                'button',
+                {
+                    type: 'button',
+                    class: 'btn btn-primary col-2',
+                },
+                {
+                    click: async () => {
+                        localStorage.setItem(WEATHER_COUNTRY_KEY, this._countrySearchElement.value);
+                        await this._refreshWeatherContainer();
+                    },
+                },
+                'Save'
+            )
+        );
+    }
+
+    async _refreshWeatherContainer() {
+        try {
+            await this._fetchData();
+        } catch (error) {
+            new Notification().showError('Fetch weather data error', error);
+        }
+        this._createWeatherContainer();
```

Comment:
I think this one will fire before you reload wather data. you should have put it in try block under await this._fetchData()